server/ec2spotmanager/tasks.py

```diff
@@ -6,19 +6,18 @@
 import boto.exception
 import fasteners
 from django.utils import timezone
-from laniakea.core.manager import Laniakea
-from laniakea import LaniakeaCommandLine
+from laniakea.core.userdata import UserData
+from laniakea.core.providers.azure.manager import AzureManager
+from laniakea.core.providers.ec2.manager import EC2Manager
+from laniakea.core.providers.ec2 import Ec2CommandLine
 from celeryconf import app
 from . import cron  # noqa ensure cron tasks get registered
 from .common.prices import get_spot_prices, get_price_median
 
-
 logger = logging.getLogger("ec2spotmanager")
 
-
 SPOTMGR_TAG = "SpotManager"
 
-
 @app.task
 def check_instance_pool(pool_id):
```

Comment:
There needs to be some abstraction here. This task has parts which will be generic for an `InstancePool` of any provider, and parts which are EC2 specific.